core/services/keeper/orm_test.go

```diff
@@ -135,6 +136,49 @@ func TestKeeperDB_BatchDeleteUpkeepsForJob(t *testing.T) {
 	require.Equal(t, int64(1), remainingUpkeep.UpkeepID)
 }
 
+func TestKeeperDB_EligibleUpkeeps_Shuffle(t *testing.T) {
+	t.Parallel()
+	db, config, orm := setupKeeperDB(t)
+	ethKeyStore := cltest.NewKeyStore(t, db, config).Eth()
+
+	blockheight := int64(63)
+	gracePeriod := int64(10)
+
+	registry, _ := cltest.MustInsertKeeperRegistry(t, db, orm, ethKeyStore)
+
+	upkeeps := [10]keeper.UpkeepRegistration{
+		newUpkeep(registry, 0),
+		newUpkeep(registry, 1),
+		newUpkeep(registry, 2),
+		newUpkeep(registry, 3),
+		newUpkeep(registry, 4),
+		newUpkeep(registry, 5),
+		newUpkeep(registry, 6),
+		newUpkeep(registry, 7),
+		newUpkeep(registry, 8),
+		newUpkeep(registry, 9),
+	}
+
+	for _, upkeep := range upkeeps {
+		err := orm.UpsertUpkeep(&upkeep)
+		require.NoError(t, err)
+	}
+
+	cltest.AssertCount(t, db, "upkeep_registrations", 10)
+
+	eligibleUpkeeps, err := orm.EligibleUpkeepsForRegistry(registry.ContractAddress, blockheight, gracePeriod)
+	assert.NoError(t, err)
+
+	require.Len(t, eligibleUpkeeps, 10)
+	ordered := [10]int64{}
+	shuffled := [10]int64{}
+	for i := 0; i < 10; i++ {
+		ordered[i] = upkeeps[i].UpkeepID
+		shuffled[i] = eligibleUpkeeps[i].UpkeepID
+	}
+	assert.NotEqualValues(t, ordered, shuffled)
```

Comment:
10 might be a kind of small number for this test. I bet with all our CI runs we will see it fail at least once. Can we decrease the probability of seeing a false positive? Or Make it more deterministic somehow?

Reply:
sure ill up it to 100 so odds should be low